### ISSUE TITLE
Airflow pipeline ingestion failures with a task from another DAG

### ISSUE THREAD

**Issue by ajsquared** [open-metadata/OpenMetadata#17075]
**Affected module**
Ingestion Framework

**Describe the bug**
We get failures in ingesting Airflow piplelines like this:

```
[2024-07-15, 08:00:40 EDT] {topology_runner.py:240} WARNING - Unexpected value error when processing stage: [type_=<class 'metadata.generated.schema.entity.data.pipeline.Pipeline'> processor='yield_pipeline' nullable=False must_return=False overwrite=True consumer=['pipeline_service'] context='pipeline' store_all_in_context=False clear_context=False store_fqn=False cache_entities=False use_cache=True]: 1 validation error for Pipeline
tasks -> 0 -> owner -> type
  field required (type=value_error.missing)
[2024-07-15, 08:00:40 EDT] {status.py:76} WARNING - Failed to ingest Pipeline Status [Airflow.airflow_dag_name] due to api request failure: Invalid task name task_from_another_dag
[2024-07-15, 08:00:40 EDT] {status.py:76} WARNING - Failed to ingest Pipeline Status [Airflow.airflow_dag_name] due to api request failure: Invalid task name task_from_another_dag
[2024-07-15, 08:00:40 EDT] {status.py:76} WARNING - Failed to ingest Pipeline Status [Airflow.airflow_dag_name] due to api request failure: Invalid task name task_from_another_dag
[2024-07-15, 08:00:40 EDT] {status.py:76} WARNING - Failed to ingest Pipeline Status [Airflow.airflow_dag_name] due to api request failure: Invalid task name task_from_another_dag
[2024-07-15, 08:00:40 EDT] {status.py:76} WARNING - Failed to ingest Pipeline Status [Airflow.airflow_dag_name] due to api request failure: Invalid task name task_from_another_dag
[2024-07-15, 08:00:40 EDT] {status.py:76} WARNING - Failed to ingest Pipeline Status [Airflow.airflow_dag_name] due to api request failure: Invalid task name task_from_another_dag
[2024-07-15, 08:00:40 EDT] {status.py:76} WARNING - Failed to ingest Pipeline Status [Airflow.airflow_dag_name] due to api request failure: Invalid task name task_from_another_dag
[2024-07-15, 08:00:40 EDT] {status.py:76} WARNING - Failed to ingest Pipeline Status [Airflow.airflow_dag_name] due to api request failure: Invalid task name task_from_another_dag
[2024-07-15, 08:00:40 EDT] {status.py:76} WARNING - Failed to ingest Pipeline Status [Airflow.airflow_dag_name] due to api request failure: Invalid task name task_from_another_dag
[2024-07-15, 08:00:40 EDT] {status.py:76} WARNING - Failed to ingest Pipeline Status [Airflow.airflow_dag_name] due to api request failure: Invalid task name task_from_another_dag
```

```
[2024-07-16, 09:00:45 EDT] {topology_runner.py:239} DEBUG - Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/api/topology_runner.py", line 237, in _process_stage
    yield from self.sink_request(stage=stage, entity_request=entity_request)
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/api/topology_runner.py", line 499, in sink_request
    yield from self.yield_and_update_context(
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/api/topology_runner.py", line 380, in yield_and_update_context
    entity = self.metadata.get_by_name(
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/ometa/ometa_api.py", line 293, in get_by_name
    return self._get(
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/ometa/ometa_api.py", line 337, in _get
    return entity(**resp)
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for Pipeline
tasks -> 0 -> owner -> type
  field required (type=value_error.missing)
[2024-07-16, 09:00:45 EDT] {topology_runner.py:240} WARNING - Unexpected value error when processing stage: [type_=<class 'metadata.generated.schema.entity.data.pipeline.Pipeline'> processor='yield_pipeline' nullable=False must_return=False overwrite=True consumer=['pipeline_service'] context='pipeline' store_all_in_context=False clear_context=False store_fqn=False cache_entities=False use_cache=True]: 1 validation error for Pipeline
tasks -> 0 -> owner -> type
  field required (type=value_error.missing)
[2024-07-16, 09:00:45 EDT] {topology_runner.py:231} DEBUG - Processing stage: type_=<class 'metadata.ingestion.models.pipeline_status.OMetaPipelineStatus'> processor='yield_pipeline_status' nullable=True must_return=False overwrite=True consumer=['pipeline_service'] context=None store_all_in_context=False clear_context=False store_fqn=False cache_entities=False use_cache=False
[2024-07-16, 09:00:45 EDT] {status.py:76} WARNING - Failed to ingest Pipeline Status [Airflow.airflow_dag_name] due to api request failure: Invalid task name task_from_another_dag
[2024-07-16, 09:00:45 EDT] {status.py:77} DEBUG - Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/ometa/client.py", line 219, in _one_request
    resp.raise_for_status()
  File "/home/airflow/.local/lib/python3.10/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://openmetadata-prod.openmetadata.svc.cluster.local:8585/api/v1/pipelines/Airflow.airflow_dag_name/status
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/sink/metadata_rest.py", line 145, in _run
    return self._run_dispatch(record)
  File "/usr/local/lib/python3.10/functools.py", line 926, in _method
    return method.__get__(obj, cls)(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/sink/metadata_rest.py", line 355, in write_pipeline_status
    pipeline = self.metadata.add_pipeline_status(
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/ometa/mixins/pipeline_mixin.py", line 44, in add_pipeline_status
    resp = self.client.put(
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/utils/execution_time_tracker.py", line 195, in inner
    result = func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/ometa/client.py", line 298, in put
    return self._request("PUT", path, data)
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/ometa/client.py", line 193, in _request
    return self._one_request(method, url, opts, retry)
  File "/home/airflow/.local/lib/python3.10/site-packages/metadata/ingestion/ometa/client.py", line 237, in _one_request
    raise APIError(error, http_error) from http_error
metadata.ingestion.ometa.client.APIError: Invalid task name task_from_another_dag
```

**To Reproduce**
I'm not sure what triggers this behavior as most of our DAGs are ingested fine. `task_from_another_dag` does exist, it is just not in  `airflow_dag_name` and has never been.

**Expected behavior**

**Version:**
 - OS: Linux
 - Python version: 3.10
 - OpenMetadata version: 1.4.3
 - OpenMetadata Ingestion package version: 1.4.3.2

**Additional context**
The Airflow instance we're ingesting from is on 2.8.3

**Comment by SumanMaharana** [comment]
Hello @ajsquared, here the error indicates that the type `field` within `owner` is missing. Could you please provide the value that is being passed into the `owner` field?

**Comment by ajsquared** [comment]
We set it in the `default_args` of the DAG:


```python
with DAG(
    "airflow_dag_name",
    default_args={
        "owner": "owner-team-name",
        "depends_on_past": False,
        "email_on_failure": False,
        "email_on_retry": False,
        "retries": 0,
        "retry_delay": timedelta(minutes=5),
    },
    description="A simple example DAG",
    schedule=timedelta(hours=1),
    start_date=datetime(2024, 4, 1),
) 
```

**Comment by Nikita054** [comment]
The same error occurs after running metadata ingestion for Airflow after couple days of successful runs.

After research we noticed that setting numberOfStatus = 0(instead of 10 by default) fixes the problem, but it works only after recreating Pipeline Service for Airflow.
Seems like something goes wrong with it.

**Comment by ajsquared** [comment]
I've noticed an issue when calling `api/v1/pipelines/{name}` (https://github.com/open-metadata/OpenMetadata/blob/main/openmetadata-service/src/main/java/org/openmetadata/service/resources/pipelines/PipelineResource.java#L227).

If the tasks in the pipeline have an owner, the `owner` is missing the `type` field that is required in `EntityReference`, e.g.
```
tasks": [
    {
      "name": "task_name",
      "fullyQualifiedName": "Airflow.dag_name.task_name",
      "sourceUrl": "<redacted>",
      "downstreamTasks": [],
      "taskType": "PythonOperator",
      "tags": [],
      "owner": {
        "id": "004c3e3d-fc4d-4ec5-9273-0d0cb4262928",
        "name": "name",
        "fullyQualifiedName": "name",
        "displayName": "name",
        "deleted": false
      }
    },
```

I think this is what triggers this error in the ingestion
```
[2024-07-15, 08:00:40 EDT] {topology_runner.py:240} WARNING - Unexpected value error when processing stage: [type_=<class 'metadata.generated.schema.entity.data.pipeline.Pipeline'> processor='yield_pipeline' nullable=False must_return=False overwrite=True consumer=['pipeline_service'] context='pipeline' store_all_in_context=False clear_context=False store_fqn=False cache_entities=False use_cache=True]: 1 validation error for Pipeline
tasks -> 0 -> owner -> type
  field required (type=value_error.missing)
```

I believe the `Pipeline` is then missing from the context, causing other issues downstream.

**Comment by SumanMaharana** [comment]
Hello @ajsquared,
I wanted to check if you've followed these specific steps in your process:

1. Created a user who owns the task in OM.
2. Ingested Airflow tasks with that user as the owner.
3. Soft-deleted the user.
4. Ingested the Airflow tasks again.

Have you done something similar?

**Comment by ajsquared** [comment]
No, I haven't done anything like that. None of the users have been soft or hard deleted.

**Comment by SumanMaharana** [comment]
Hello @ajsquared, how are your running the ingestion

**Comment by ajsquared** [comment]
Nothing out of the ordinary - the ingestion is configured in the OpenMetadata API and deployed to the Airflow instance installed by the openmetadata-dependencies Helm chart.

**Comment by SumanMaharana** [comment]
Hello @ajsquared, can you test it out with the latest version (1.5.0) of Openmetadata. I believe we have fixed various things on owners.
Its not reproducible unless we take the below steps which is fixed on the latest version (1.5.0) :

> Hello @ajsquared, I wanted to check if you've followed these specific steps in your process:
> 
> 1. Created a user who owns the task in OM.
> 2. Ingested Airflow tasks with that user as the owner.
> 3. Soft-deleted the user.
> 4. Ingested the Airflow tasks again.
> 
> Have you done something similar?

Closing this issue. Thanks

**Comment by FFCMSouza** [comment]
It still occurs on Openmetadata 1.8. When a dag used to have a task and this tasks does not exist anymore the error appear. I don't see any workaround.
